commit ee5185926e8758610234acb79c039d4b32afb051
Author: Miroslav Rezanina <mrezanin@redhat.com>
Date:   Mon Mar 4 09:16:37 2013 +0100

    char: Disable write callback if throttled chardev is  detached
    
    Message-Id: <736b3ad749e5c60de44d4c5836b59d2287b6b918.1354903384.git.crobinso@redhat.com>
    Bugzilla: 745758
    Upstream: Not applicable - rhel only
    
    If a throttled chardev is detached from the frontend device, all future
    callbacks should be suppressed.  Not doing this results in a segfault.
    
    Signed-off-by: Amit Shah <amit.shah@redhat.com>
    Signed-off-by: Cole Robinson <crobinso@redhat.com>

diff --git a/qemu-char.c b/qemu-char.c
index 52ddc16..9f9b11e 100644
--- a/qemu-char.c
+++ b/qemu-char.c
@@ -223,6 +223,11 @@ void qemu_chr_add_handlers(CharDriverState *s,
         ++s->avail_connections;
     }
     if (!handlers) {
+        if (s->write_blocked) {
+            /* Ensure we disable the callback if we were throttled */
+            s->chr_disable_write_fd_handler(s);
+            /* s->write_blocked is cleared below */
+        }
         handlers = &null_handlers;
     }
     s->chr_can_read = handlers->fd_can_read;
